internal/handlers/pg/pgdb/pool.go

```diff
@@ -291,10 +291,14 @@ func (pgPool *Pool) CreateDatabase(ctx context.Context, db string) error {
 	}
 	defer func() {
 		if err != nil {
-			pgPool.logger.Error("failed to perform rollback", zap.Error(tx.Rollback(ctx)))
+			if rerr := tx.Rollback(ctx); rerr != nil {
+				pgPool.logger.Error("failed to perform rollback", zap.Error(rerr))
+			}
 			return
 		}
-		pgPool.logger.Error("failed to perform commit", zap.Error(tx.Commit(ctx)))
+		if cerr := tx.Commit(ctx); cerr != nil {
+			pgPool.logger.Error("failed to perform commit", zap.Error(cerr))
```

Comment:
I think `CreateDatabase` should error in that case. That would require a named result, etc.

And I think we should call `Rollback` there too

Reply:
Yeah, good point...